libs/src/services/contracts/ContractClient.js

```diff
@@ -78,13 +78,15 @@ class ContractClient {
         return
       }
 
-      await this.retryInit()
+      await this.retryInit(selectNewEndpointOnRetry)
     }
   }
 
-  async retryInit () {
+  async retryInit (selectNewEndpoint = true) {
```

Comment:
i can go either way, but i do like this being default to true

Reply:
basically
true - optimized for client availability
false - optimized for edge cases/local dev/rollout

right?

Reply:
yeah so this isn't the default value that hareesh and i were discussing. that is in `dataContracts/index.js:initUserReplicaSetManagerClient()` function. since the `contractClient` code change impacts every contract init for every user, i wanted to set the default to maintain current functionality